WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -29,6 +29,9 @@ class CompetitionsController < ApplicationController
 
   def new
     @competition = Competition.new
+    if current_user.any_kind_of_delegate?
+      @competition.delegates |= [current_user]
```

Comment:
Came up with the same thought but it shouldn't. When using `||=` we expect that `@competition.delegates` is nil, whereas it's `#<ActiveRecord::Associations::CollectionProxy []>`. So we need to get the union here.